rmf_fleet_adapter/src/rmf_fleet_adapter/agv/test/MockAdapter.cpp

```diff
@@ -70,13 +70,15 @@ class MockScheduleNode : public rmf_traffic::schedule::Writer
   {
     _worker.schedule(
       [
-        database = _database,
+        database = std::weak_ptr<rmf_traffic::schedule::Database>(_database),
```

Comment:
I would argue that the original use of `std::shared_ptr` here is fine. It just means the database will stay alive until the worker gets around to performing this operation. I don't believe there's any circular dependency issue here.

Same for the other database operations.

Reply:
You're right. I didn't notice that worker was owned by `MockAdapter` :man_facepalming: 703a3b5

Reply:
My understanding of "workers" in RxCpp is that they're kind-of-sort-of global objects, so they aren't really "owned" in the usual sense. Whatever gets scheduled to them will go into a globally stored pool of scheduled events, and get executed eventually. Once the scheduled event is executed, the lambda will get released, and everything in its capture list gets destructed. So we shouldn't really have to worry about circularity when it comes to tasks that are scheduled to workers.

That being said, I may have some misunderstandings on this since RxCpp is quite complex, so if anyone knows better please do correct me on this.